### ISSUE TITLE
Horizontal SegmentedControl is broken with right-to-left layout

### ISSUE THREAD

**Issue by Eldemarkki** [mantinedev/mantine#5002]
### What package has an issue

@mantine/core

### Describe the bug

When I have right-to-left mode enabled in Firefox 118, the horizontal SegmentedControl component is broken. The initial view looks fine, but when I click the controls, the handle moves to the wrong position.
![image](https://github.com/mantinedev/mantine/assets/38920928/db856dae-782b-406d-a9d0-f867f8c0bf4b)

![image](https://github.com/mantinedev/mantine/assets/38920928/71020b0f-2362-49bc-aaf5-fbb892564318)

![image](https://github.com/mantinedev/mantine/assets/38920928/03f8c326-29e6-4962-ac8f-b5f0249b63c1)

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.1.2

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by Eldemarkki** [comment]
This might actually just be a bug with the docs website

**Comment by rtivital** [comment]
I cannot reproduce the issue, it works fine on the docs website. Make sure that you've followed rtl setup documnetation.

https://github.com/mantinedev/mantine/assets/10353856/8ebfc4ff-407f-46a8-8774-753824f72495

**Comment by Eldemarkki** [comment]
I just noticed this bug while playing around with the docs page, maybe I should've used the docs issue template, sorry for the confusion. I don't use/need RTL in my own project. 

The bug seems to only happen on Firefox when switching direction with Firefox's built-in "Switch Page Direction" feature:
![image](https://github.com/mantinedev/mantine/assets/38920928/d5b4629f-ef5b-4173-a37e-ea7595e59b60)

Everything works correctly when I switch the direction with the button in the navbar:
![image](https://github.com/mantinedev/mantine/assets/38920928/6f3a85a1-bb83-4eb6-b4e4-2c65ca00409e)

I noticed that the RTL button in the navbar doesn't update when clicking Firefox's "Switch Page Direction", so the button or `DirectionProvider` gets out-of-sync with the actual direction. Maybe MutationObserver could be used for listening to page direction change?
